cpp/kiss_icp/core/Preprocessing.cpp

```diff
@@ -43,9 +43,11 @@ std::vector<Eigen::Vector3d> VoxelDownsample(const std::vector<Eigen::Vector3d>
         const auto voxel = PointToVoxel(point, voxel_size);
         if (!grid.contains(voxel)) grid.insert({voxel, point});
     });
-    std::vector<Eigen::Vector3d> frame_dowsampled(grid.size());
-    std::transform(grid.cbegin(), grid.cend(), frame_dowsampled.begin(),
-                   [](const auto &voxel_and_point) { return voxel_and_point.second; });
+    std::vector<Eigen::Vector3d> frame_dowsampled;
+    frame_dowsampled.reserve(grid.size());
+    std::for_each(grid.cbegin(), grid.cend(), [&](const auto &voxel_and_point) {
```

Comment:
@tizianoGuadagnino commenting here as the other thread is too large now

I think, overall, this is the best approach, as we mainly use for_each everywhere in the code. It might be too much for newbies, but this also forces them to read what is the algo doing, as it's now everywhere in the code

Let's keep this double for each implementation, is short and easier to read than `transform, make, my mandess`

Reply:
Awesome, I really like this new version.